.github/workflows/daily.yml

```diff
@@ -34,7 +34,10 @@ jobs:
         uses: actions/setup-python@v4
         with:
           python-version: ${{ matrix.python-version }}
+          cache: 'pip'
+          cache-dependency-path: requirements-tests.txt
       - name: Update pip
+        if: matrix.os == "macos-11"
```

Comment:
Why do we need this line? I usually update pip on all envs.

Reply:
> Usually, pip is always updated and included in [runner images](https://github.com/actions/runner-images/blob/main/images/linux/Ubuntu2204-Readme.md?rgh-link-date=2022-09-29T16%3A30%3A56Z). Because of this, the latest version of pip is usually installed already.

https://github.com/actions/setup-python/issues/506#issuecomment-1262522726

I can only assume `python -m pip install -U pip` was done because of `"macos-11"`. As the only times that command is ran is when also using `"macos-11`. So only running it on that OS is a small optimization for the others.

Reply:
Given that we have `PIP_DISABLE_PIP_VERSION_CHECK: 1` set as an environment variable in this workflow, we probably don't need this step at all. It usually isn't essential to have the very latest version of pip installed; the only significant benefit from upgrading pip in CI is that you don't get the annoying "WARNING: a newer version of pip is available, you should upgrade NOW" message with every CI job.

Reply:
I didn't even notice `PIP_DISABLE_PIP_VERSION_CHECK`. Should I remove the step? Leave it as it was before?

Reply:
I reckon you can remove it. There's other workflows we have in this repo where we don't have it as a step, and it fits with the theme of this PR ("speed up CI by getting pip to do less work").